### ISSUE TITLE
Vaadin 14 dev mode not compatible with newer node and npm versions

### ISSUE THREAD

**Issue by futaz** [vaadin/flow#12546]
### Describe the bug

I have newer node and npm versions installed on my system (see System info section). When I download a new vaadin 14.7.8 app from vaadin starter, after compiling and deploying in **development mode** I get an Internal Server Error. 
```
2021-12-08 12:27:08.106 ERROR 73821 --- [nio-8080-exec-2] c.v.flow.server.DefaultErrorHandler      : 

com.vaadin.flow.server.BootstrapException: Unable to read webpack stats file.
	at com.vaadin.flow.server.BootstrapHandler$BootstrapPageBuilder.setupFrameworkLibraries(BootstrapHandler.java:956) ~[flow-server-2.7.6.jar:2.7.6]
	at com.vaadin.flow.server.BootstrapHandler$BootstrapPageBuilder.setupDocumentHead(BootstrapHandler.java:833) ~[flow-server-2.7.6.jar:2.7.6]
```
For more detais see attached [Stack trace](https://github.com/vaadin/platform/files/7676713/vaadin-starter-14.7.8-01-1-initial-log.txt).

### Workaround

install node and npm in project folder with [frontend-maven-plugin](https://github.com/eirslett/frontend-maven-plugin#installing-node-and-npm) with for example `<nodeVersion>v12.22.7</nodeVersion>` and `<npmVersion>6.14.15</npmVersion>` (see attached [pom.xml](https://github.com/vaadin/platform/files/7676723/pom.xml.txt)) then rebuild and deploy the project.

With older node and npm the development mode and webpack dev server works well.

### Fix

Please make the vaadin dev mode work on new node/npm versions or communicate the supported versions in documentation and in the generated README.md or generate the projects with the frontend-maven-plugin in the pom.xml.

### Expected-behavior

Vaadin app opens in browser without internal server error.

### Reproduction

- Download a V14 LTS vaadin app with default settings from https://start.vaadin.com/app/
- unzip
- open folder in terminal
- run `./mvnw`
- when compiled and deployed, browser will open at http://localhost:8080/
- browser shows Internal Server Error with a stacktrace
- see logs for details

### System Info

- macOS Catalina 10.15.7
- node 17.2.0 (from homebrew)
- npm 8.1.4 (from homebrew)
- maven 3.8.4 (from homebrew)

**Comment by ZheSun88** [comment]
Hi @futaz , 

thanks for reporting this issue, we have fixed some issues regarding node 17. and now this looks another issue. 
we are going to investigate more and fix it as soon as possible.

thanks

**Comment by caalador** [comment]
Note for the suggested workaround. Instead of using the frontend-maven-plugin to install a local node to the project directory just add to the pom.xml vaadin-maven-plugin definition the configuration:
```
                <configuration>
                    <requireHomeNodeExec>true</requireHomeNodeExec>
                </configuration>
```
Which will then use a compatible node from `~/.vaadin` or if none exists install it there.

**Comment by joheriks** [comment]
It seems the `webpack-dev-server` version (`3.11.0`) used by V14 when run under Node 17 for some reason refuses the connection to `http://localhost:12345/assetsByChunkName` (replace `12345` with the random port assigned by `DevModeHandler`). Connection is refused when requested from Java  despite the same URL working fine when requested from browser and curl. This is not the case with the `webpack-dev-server`  (`4.1.1`) used in newer Vaadin versions.

By testing, I discovered another workaround is to add 
```
vaadin.devmode.webpack.options = --host=127.0.0.1 -d --inline=false
```

to `application.properties` when using Node 17 and developing locally.

**Comment by joheriks** [comment]
Fix #12710 coming in next LTS maintenance update.
